overops-functions/src/main/java/com/takipi/udf/microsoftteams/MicrosoftTeamsUtil.java

```diff
@@ -0,0 +1,79 @@
+package com.takipi.udf.microsoftteams;
+
+import com.google.common.base.Strings;
+import com.takipi.api.client.ApiClient;
+import com.takipi.api.client.data.service.SummarizedService;
+import com.takipi.api.client.data.view.SummarizedView;
+import com.takipi.api.client.util.client.ClientUtil;
+import com.takipi.api.client.util.view.ViewUtil;
+import com.takipi.udf.ContextArgs;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import java.util.List;
+
+public class MicrosoftTeamsUtil {
+
+    public static EventData getEventData(ApiClient apiClient, ContextArgs args) {
```

Comment:
I don't understand the purpose of this class.
It seems like `servers`/`applications`/`deployments` are only used in the context of `EventUtil.getEventRecentLink`

However, giving that method all of them is the equivalent of giving that method none of them.
If none are passed, the api server won't filter the snapshot, which is the same as asking it to filter based on all of them.

Also, the process of calling `ClientUtil.getServers`, `ClientUtil.getApplications` and `ClientUtil.getDeployments` is time consuming for the UDF, and resource consuming for the API server.

Regarding the environments, there's now `ClientUtil.getEnvironment(apiClient, serviceId)` which can be used.

Reply:
Done